freqtrade/persistence.py

```diff
@@ -331,15 +340,15 @@ def close(self, rate: float) -> None:
             self
         )
 
-    def calc_open_trade_price(self, fee: Optional[float] = None) -> float:
+    def _calc_open_trade_price(self) -> float:
         """
         Calculate the open_rate including fee.
         :param fee: fee to use on the open rate (optional).
         If rate is not set self.fee will be used
         :return: Price in of the open trade incl. Fees
         """
         buy_trade = (Decimal(self.amount) * Decimal(self.open_rate))
-        fees = buy_trade * Decimal(fee or self.fee_open)
+        fees = buy_trade * Decimal(self.fee_open)
```

Comment:
maybe not for this PR, maybe fits here:
* do we really need Decimals here and in `calc_close_trade_price()`?
* redundant parentheses one line above (`buy_trade = (Decimal(self.amount) * Decimal(self.open_rate))`)

Reply:
I agree with both points (removed the paranteses now) ... 
and i've questioned these "Decimal" conversations many times before ... but i've never come to a clear conlusion. 

Point is - it breaks a few tests (specifically, tests checking for numbers ...) - which i don't see as a "good" sign if that's mixed into a PR which refactors the calculation logic - so i would definitely leave this out for now.

Reply:
Okay, agree, let's look at them (the Decimals here) separately if they break the tests.
Removing them should speedup a bit these functions frequently called in backtesting...

Reply:
yeah, one of the breaks i looked at changed the 16th (probably) decimal place ... 
so from `0.0010024999999225068` to `0.0010024999999225066` ... 

but as said - i find if i refactor a function, and the refactored calculation causes (even so slightly) different numbers, it leaves a odd feeling about that PR ... ;) 

doesn't break too many tests though - just 5 or 6